### PR TITLE
DMP-161 Create REST Endpoint to receive Events request

### PR DESCRIPTION
**Pull request by marcmollins**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-161

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/Events/controller/EventsController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import javax.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+@Slf4j
+@RestController
+public class EventsController implements EventApi {
+
+    @Operation(
+        operationId = "eventAddDocumentPost",
+        summary = "SORT OUT LATER.",
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/darts/Events/controller/EventsController.java:27]: Can you add a summary to this?

File: src/test/java/uk/gov/hmcts/darts/Events/controller/EventsControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                .queryParam("message_id", "400")
+                .queryParam("type", "1")
+                .queryParam("sub_type", "1")
+                .content(requestBody);
+            MvcResult response = mockMvc.perform(requestBuilder).andExpect(status().isNotImplemented()).andReturn();
+
+            assertThat(response.getResponse().getContentAsString()).isEqualTo("");
+        }
+
+//        @Test
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/darts/Events/controller/EventsControllerTest.java:47]: Remove commented out ode

File: src/main/resources/openapi/event.yaml [hunk 3]
```diff
@@ -16,27 +16,27 @@ paths:
     post:
       tags:
         - Event
-      summary: Event data received from XHIBIT and CPP through a custom web service to add context to the audio recordings it stores, which allows users to request audio by case and hearing data without needing to know the time or physical location of where the audio was recorded.
+      summary: An Endpoint which allows users to request audio by case and hearing data
```
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/event.yaml:19]: i think this is a generic endpoint for an event coming in, its not specific to audio.
/addAudio would be specifically for this

File: src/main/java/uk/gov/hmcts/darts/Events/controller/EventsController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.darts.Events.controller;
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/Events/controller/EventsController.java]: Events should have a small e at the start

**Review by cmagowan-hmcts — APPROVED** [review]

**Pull request merged**